[vweb] string as response and use text/plain content type but binary return

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:**  V 0.2.4 fb3a793
**OS:** Darwin

**What did you do?**
```
import vweb

struct App {
    vweb.Context
}

fn (mut app App) remote_ip() vweb.Result {
	return app.text(app.ip())
}

fn main() {
	vweb.run(&App{}, 8080)
}
```
curl http://localhost:8080/remote_ip

**What did you expect to see?**
192.168.1.1
 
**What did you see instead?**
❯ curl -X GET http://localhost:8080/remote_ip
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.

on MacOS, having used brew to install openssl, the code above does not compile

```
user: thomas ~/ via V v0.2.4 
at 17:12:58 ❯ v -cg run  test.v
/tmp/v_501/test.10696759268661050370.tmp.c:14606:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t15.data);
   ^~~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:14786:4: warning: expression result unused [-Wunused-value]
  (*(string*)_t1.data);
   ^~~~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:14895:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t1.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:14908:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t1.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:15080:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t1.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:15108:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t5.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:17029:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t7.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:17055:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t13.data);
   ^~~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:17500:4: warning: expression result unused [-Wunused-value]
  (*(string*)_t1.data);
   ^~~~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:17690:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t2.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:17715:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t2.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:18942:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t2.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21158:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t1.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21199:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t1.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21226:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t6.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21513:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21539:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21549:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t3.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21559:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t5.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21920:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21978:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t5.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:21986:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t7.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22099:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t5.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22139:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t4.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22158:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22171:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22183:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22220:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t3.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22234:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t6.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22269:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22389:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t4.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22602:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t7.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22614:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t9.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22665:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:22678:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:23059:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:23069:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t3.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:23079:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t5.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:24899:4: warning: expression result unused [-Wunused-value]
  (*(bool*)_t7.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:25040:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t5.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:25433:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t5.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:25634:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t3.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:26472:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t8.data);
   ^~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:26655:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t21.data);
   ^~~~~~~~~~~~~~~~
/tmp/v_501/test.10696759268661050370.tmp.c:26849:4: warning: expression result unused [-Wunused-value]
  (*(int*)_t1.data);
   ^~~~~~~~~~~~~~~
45 warnings generated.
ld: warning: directory not found for option '-L/opt/local/lib'
ld: warning: directory not found for option '-L/usr/local/opt/openssl/lib'
ld: warning: object file (/Users/thomas/.vmodules/cache/48/48928e7c770c814d28504ffe330ad631.o) was built for newer macOS version (12.0) than being linked (10.7)
ld: warning: ignoring file /opt/homebrew/Cellar/openssl@3/3.0.0_1/lib/libssl.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
ld: warning: ignoring file /opt/homebrew/Cellar/openssl@3/3.0.0_1/lib/libcrypto.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
Undefined symbols for architecture x86_64:
  "_BIO_ctrl", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_BIO_free_all", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_BIO_new_ssl_connect", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_BIO_puts", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_BIO_read", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_OPENSSL_init_ssl", referenced from:
      _net__openssl__init in test-9a6825.o
  "_SSL_CTX_free", referenced from:
      _net__openssl__SSLConn_shutdown in test-9a6825.o
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_CTX_load_verify_locations", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_CTX_new", referenced from:
      _net__openssl__SSLConn_connect in test-9a6825.o
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_CTX_set_options", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_CTX_set_verify_depth", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_CTX_use_PrivateKey_file", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_CTX_use_certificate_file", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_connect", referenced from:
      _net__openssl__SSLConn_connect in test-9a6825.o
  "_SSL_ctrl", referenced from:
      _net__openssl__SSLConn_connect in test-9a6825.o
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_free", referenced from:
      _net__openssl__SSLConn_shutdown in test-9a6825.o
  "_SSL_get1_peer_certificate", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_get_error", referenced from:
      _net__openssl__ssl_error in test-9a6825.o
  "_SSL_get_verify_result", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_new", referenced from:
      _net__openssl__SSLConn_connect in test-9a6825.o
  "_SSL_read", referenced from:
      _net__openssl__SSLConn_socket_read_into_ptr in test-9a6825.o
  "_SSL_set_cipher_list", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_SSL_set_fd", referenced from:
      _net__openssl__SSLConn_connect in test-9a6825.o
  "_SSL_shutdown", referenced from:
      _net__openssl__SSLConn_shutdown in test-9a6825.o
  "_SSL_write", referenced from:
      _net__openssl__SSLConn_write in test-9a6825.o
  "_TLS_client_method", referenced from:
      _net__openssl__SSLConn_connect in test-9a6825.o
  "_TLS_method", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
  "_X509_free", referenced from:
      _net__http__Request_ssl_do in test-9a6825.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/net/address.v
+++ b/vlib/net/address.v
@@ -66,30 +66,30 @@ const (
 )
 
 fn (a Ip) str() string {
-	buf := []byte{len: net.max_ip_len, init: 0}
+	buf := [net.max_ip_len]char{}
 
-	res := &char(C.inet_ntop(.ip, &a.addr, buf.data, buf.len))
+	res := &char(C.inet_ntop(.ip, &a.addr, &buf[0], buf.len))
 
 	if res == 0 {
 		return '<Unknown>'
 	}
 
-	saddr := unsafe { (&byte(buf.data)).vstring() }
+	saddr := unsafe { cstring_to_vstring(&buf[0]) }
 	port := C.ntohs(a.port)
 
 	return '$saddr:$port'
 }
 
 fn (a Ip6) str() string {
-	buf := []byte{len: net.max_ip6_len, init: 0}
+	buf := [net.max_ip_len]char{}
 
-	res := &char(C.inet_ntop(.ip6, &a.addr, buf.data, buf.len))
+	res := &char(C.inet_ntop(.ip6, &a.addr, &buf[0], buf.len))
 
 	if res == 0 {
 		return '<Unknown>'
 	}
 
-	saddr := unsafe { (&byte(buf.data)).vstring() }
+	saddr := unsafe { cstring_to_vstring(&buf[0]) }
 	port := C.ntohs(a.port)
 
 	return '[$saddr]:$port'

--- a/vlib/net/address.v
+++ b/vlib/net/address.v
@@ -74,7 +74,7 @@ fn (a Ip) str() string {
 		return '<Unknown>'
 	}
 
-	saddr := buf.bytestr()
+	saddr := unsafe { (&byte(buf.data)).vstring() }
 	port := C.ntohs(a.port)
 
 	return '$saddr:$port'
@@ -89,7 +89,7 @@ fn (a Ip6) str() string {
 		return '<Unknown>'
 	}
 
-	saddr := buf.bytestr()
+	saddr := unsafe { (&byte(buf.data)).vstring() }
 	port := C.ntohs(a.port)
 
 	return '[$saddr]:$port'

--- a/vlib/net/address_test.v
+++ b/vlib/net/address_test.v
@@ -96,3 +96,17 @@ fn test_sizes_ipv4() {
 fn test_sizes_unix() {
 	assert sizeof(C.sockaddr_un) == sizeof(Unix) + aoffset
 }
+
+fn test_ip_str() {
+	ip := new_ip(1337, addr_ip_any).str()
+	expected := '0.0.0.0:1337'
+	assert ip.len == expected.len
+	assert ip == expected
+}
+
+fn test_ip6_str() {
+	ip := new_ip6(1337, addr_ip6_any).str()
+	expected := '[::]:1337'
+	assert ip.len == expected.len
+	assert ip == expected
+}

